server/src/main/java/org/elasticsearch/cluster/metadata/SingleNodeShutdownMetadata.java

```diff
@@ -378,6 +406,15 @@ public Builder setNodeId(String nodeId) {
             return this;
         }
 
+        /**
+         * @param nodeEphemeralId The node ephemeral ID this metadata refers to.
+         * @return This builder.
+         */
+        public Builder setNodeEphemeralId(String nodeEphemeralId) {
+            this.nodeEphemeralId = nodeEphemeralId;
+            return this;
+        }
+
```

Comment:
Is it worth adding a convenience method that takes a `DiscoveryNode` and sets both the node ID and ephemeral ID in one go? it might reduce a bit of boilerplate in cases where you have a reference to the `DiscoveryNode`?